custom_components/zaptec/api.py

```diff
@@ -399,327 +737,265 @@ async def _refresh_token(self):
+                self._const, ["Settings", "SettingIds"], device_types
+            )
+            self._cmd_ids = Account._get_remap(
+                self._const, ["Commands", "CommandIds"], device_types
+            )
 
-    async def combined_max(self):
-        return await self._send_command(10999)
+            # Commands can also be specified as lower case strings
+            self._cmd_ids.update(
+                {to_under(k): v for k, v in self._cmd_ids.items() if isinstance(k, str)}
+            )
 
-    async def state(self):
-        data = await self._account._request(f"chargers/{self.id}/state")
-        # sett_attributes need to be set before any other call.
-        self.set_attributes(data)
-        # Firmware version is called. SmartMainboardSoftwareApplicationVersion,
-        # stateid 908
-        # I couldn't find a way to see if it was up to date..
-        # maybe remove this later if it dont interest ppl.
+        # Update the state on all chargers
+        for data in self.map.values():
+            if isinstance(data, Charger):
+                await data.state()
 
-        # Fetch some additional attributes from settings
-        data = await self._account._request(f"chargers/{self.id}/settings")
-        max_current_id = str(self._account.obs.get("ChargerMaxCurrent", ''))
-        min_current_id = str(self._account.obs.get("ChargerMinCurrent", ''))
-        settings = {
-            "charger_max_current": data.get(max_current_id,{}).get("Value"),
-            "charger_min_current": data.get(min_current_id,{}).get("Value"),
-        }
-        self.set_attributes(settings)
+        self.is_built = True
 
-        if self.installation_id in self._account.map:
-            firmware_info = await self._account.charger_firmware(self.installation_id)
-            for fm in firmware_info:
-                if fm["ChargerId"] == self.id:
-                    fixed = {
-                        "current_firmware_version": fm["CurrentVersion"],
-                        "available_firmware_version": fm["AvailableVersion"],
-                        "firmware_update_to_date": fm["IsUpToDate"],
-                    }
-                    self.set_attributes(fixed)
+    def update(self, data: TDict):
+        """update for the stream. Note build has to called first."""
 
-    async def live(self):
-        # This don't seems to be documented but the portal uses it
-        # TODO check what it returns and parse it to attributes
-        return await self._account._request("chargers/%s/live" % self.id)
-
-    async def settings(self):
-        # TODO check what it returns and parse it to attributes
-        return await self._account._request("chargers/%s/settings" % self.id)
-
-    async def _send_command(self, id_):
-        cmd = "chargers/%s/SendCommand/%s" % (self.id, id_)
-        _LOGGER.debug("Calling %s", cmd)
-        return await self._account._request(cmd, method="post")
-
-    async def update(self, data):
```

Comment:
Comment out, don’t remove this method.

Reply:
I can add it back if you want @Hellowlol. I'm sorry because I deleted the function because it doesn't do anything, and it is messy to have have useless methods included. What is the intent of the function?

Reply:
Honestly, I can’t remember. I just remember that I needed it for something. I’ll get back to it when I have time

Reply:
I propose that we leave it out for now (since it does nothing) and add it later when you have time to look at it.